mono/utils/networking-posix.c

```diff
@@ -17,7 +17,7 @@
 #ifdef HAVE_SYS_IOCTL_H
 #include <sys/ioctl.h>
 #endif
-#ifdef HAVE_SYS_SOCKET_H
+#if defined(HAVE_SYS_SOCKET_H) || defined(__OpenBSD__)
```

Comment:
This might just be a bad merge.
The earlier code did not include socket.h, just in.h and had an ifdef around that. The earlier path added another include within the ifdef.

Reply:
I'm pretty certain this is rooted in an bad merge, and is not needed.

Reply:
The configure output indicates that it does find sys/socket.h, is HAVE_SYS_SOCKET_H defined in config.h ?

Reply:
HAVE_SYS_SOCKET_H is set to 1 in config.h

I looked through config.h, config.h.in, config.log, and the output of `./configure --verbose`, but failed to find anything that would explain it to me.

Here are the files if you wanna take a look:

https://gist.github.com/thfrwn/ec78cfbf680d5847e26610e64df085fe

Reply:
So HAVE_SYS_SOCKET_H is defined in config.h, and networking-posix.c includes that file in the beginning, so not sure what is happening here.

Reply:
Those lines were added by e31ecb3b4198.

Reply:
I'm doing this with the 5.10.0.179 from https://download.mono-project.com/sources/mono/ - would be surprised if a Jan 8 commit wasn't in that?

Reply:
5.10 was branched in December (2017-12 branch) so that does not surprise me :)

Reply:
did it not enter the stable build?

Reply:
We do branch for release every two months so depending on when it was committed it is included or not. It did make the 2018-02 branch which will be Mono 5.12.

Reply:
gotcha. I'll just take this patch out then